### DISCUSSION TITLE
Masking ENV variables in reusable actions written in python

### DISCUSSION THREAD

**Discussion by jason-oc** [community/community#108392] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a reusable (composite) action whose steps are written in python. The first step of this action reads secret information (from AWS Secrets Manager) and writes that information to GITHUB_ENV for use by subsequent steps in the same reusable action.

There are 2 pieces of secret information retrieved from AWS SM and written to GITHUB_ENV; a GitHub Personal Access Token (PAT) and an API key for Artifactory.

When I execute a workflow that is utilizing this reusable action, I see each step in the reusable action is printed to the workflow console log along with its environment variables. In the case of the GitHub PAT it is masked (for some reason unknown to me) but in the case of the Artifactory API key it is not masked.

My main concern is obviously how I can also mask the API key but I would also like to know how/why the GitHub PAT, which is being handled in the exact same way in my code, is already being masked?

Please keep in mind that I am using a reusable action. I have already read plenty of documentation on how this can be achieved directly in a workflow but that does not help me with my use case.

**Comment by jason-oc** [comment, marked as answer]
I am astounded, but apparently the answer is as follows:

env_file = os.environ.get('GITHUB_ENV')
print("::add-mask::" + ARTIFACTORY_API_KEY)
with open(env_file, "a") as envfile:
envfile.write("ARTIFACTORY_API_KEY=" + str(ARTIFACTORY_API_KEY) + "\n")

**Comment by finebel** [comment]
Thanks for your answer!

> but I would also like to know how/why the GitHub PAT, which is being handled in the exact same way in my code, is already being masked?

Not sure whether it's relevant anymore, but have a look at https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#redacting-secrets-from-workflow-run-logs. Among other information, v2 GitHub PATs are redacted automatically.
